crates/ruff_linter/src/rules/ruff/rules/function_call_in_dataclass_default.rs

```diff
@@ -159,7 +159,7 @@ fn any_annotated(class_body: &[Stmt]) -> bool {
         .any(|stmt| matches!(stmt, Stmt::AnnAssign(..)))
 }
 
-fn is_immutable_newtype_call(
+pub(crate) fn is_immutable_newtype_call(
```

Comment:
I think we should move this function to https://github.com/astral-sh/ruff/blob/2ef94e5f3e6e57a19151334b598e2bf381e97dac/crates/ruff_python_semantic/src/analyze/typing.rs as it's being used by rules from multiple plugins, is related to the `typing` module. We should also add docs to it.

Reply:
We should also update the rule documentation to specify this behavior regarding `NewType`.